src/bokeh/models/widgets/tables.py

```diff
@@ -858,6 +858,50 @@ def __init__(self, *args, **kwargs) -> None:
     The height of each row in pixels.
     """)
 
+    @staticmethod
+    def from_data(data, columns=None, formatters={}, **kwargs) -> DataTable:
+        """ Create a simple table from a pandas dataframe, dictionary or ColumnDataSource.
+
+        Args:
+            data (DataFrame or dict or ColumnDataSource) :
+                The data to create the table from. If the data is a dataframe
+                or dictionary, a ColumnDataSource will be created from it.
+
+            columns (list, optional) :
+                A list of column names to use from the input data.
+                If None, use all columns. (default: None)
+
+            formatters (dict, optional) :
+                A mapping of column names and corresponding Formatters to
+                apply to each column. (default: None)
+
+        Keyword arguments:
+            Any additional keyword arguments will be passed to DataTable
+
+        Returns:
+            DataTable
+
+        Raises:
+            ValueError
+                If the data is not a pandas Dataframe, dictionary or ColumnDataSource
+
+        """
+        import pandas as pd
+
+        if isinstance(data, (pd.DataFrame, dict)):
+            source = ColumnDataSource(data)
+        elif isinstance(data, ColumnDataSource):
+            source = data
+        else:
+            raise ValueError("Data should be a pandas DataFrame, dictionary, or a Bokeh ColumnDataSource.")
+
+        if columns is not None:
+            source.data = {col: source.data[col] for col in columns}
+
+        table_columns = [TableColumn(field=c, title=c, formatter=formatters.get(c, CellFormatter())) for c in source.data]
```

Comment:
`CellFormatter` is an abstract base class and you won't be able to instantiate it in bokehjs. In bokeh, well in Python, we mark it as `@abstract`, but we can't/don't make such classes' constructors fail at run-time. However, `TableColumn`'s `formatter` property has a sensible default value, so instead of providing your own, use `Intrinsic` (from `bokeh.core.properties.singletons`), which will tell the properties' system to use the default value whatever that is (can be unset for example).

Reply:
Another alternative is to not pass the `formatter` parameter at all when the `formatters` dict does not have a key for a column. This would require re-arranging the code a little, e.g making an explicit loop rather than a comprehension so that you can insert that kind of conditional.